coderd/rbac/authz.go

```diff
@@ -91,13 +93,27 @@ type authSubject struct {
 // ByRoleName will expand all roleNames into roles before calling Authorize().
 // This is the function intended to be used outside this package.
 // The role is fetched from the builtin map located in memory.
-func (a RegoAuthorizer) ByRoleName(ctx context.Context, subjectID string, roleNames []string, action Action, object Object) error {
+func (a RegoAuthorizer) ByRoleName(ctx context.Context, subjectID string, roleNames []string, scope Scope, action Action, object Object) error {
 	roles, err := RolesByNames(roleNames)
 	if err != nil {
 		return err
 	}
 
-	return a.Authorize(ctx, subjectID, roles, action, object)
+	err = a.Authorize(ctx, subjectID, roles, action, object)
+	if err != nil {
+		return err
+	}
+
+	// If the scope isn't "any", we need to check with the scope's role as well.
+	if scope != ScopeAll {
+		scopeRole := builtinScopes[scope]
```

Comment:
Might want to check `ok`. Invalid scopes should return unauthorized imo.
```suggestion
		scopeRole, ok := builtinScopes[scope]
```